CHANGELOG.rst

```diff
@@ -25,6 +25,10 @@ Breaking changes:
 * Rename ``session`` parameter to ``cacheid`` in many places
 * Rename public functions in the ``jsonpatch`` module
 
+Bug Fixes:
+
+* Resolve references to "$anchor", $dynamicAnchor", or subschema "$id" in newly loaded schemas
```

Comment:
Unless additional scenarios are clearly described in the issue/PR, then let's just summarize the fixed test cases: I think "Fixed remote anchor ref resolution" is fine here.

Reply:
Updated- I wrote "Fixed remote anchor and identifier ref resolution" beause of `"$id"` and because I'm pedantic like that, but I'm happy to change it to your text if you'd prefer (I respect the pedantic choices of others!  😜 )  Or you're welcome to edit it in the PR, whatever works.

Reply:
@marksparkza Ah, I see what you mean here - I appreciate your diligence!  You are correct that this does not fix remote `$id`s.  The remaining problem is more related to #53, which I need to close as overly muddled (I did not yet understand `Catalog`, `Source`, and how `JSONSchema` loads things thoroughly) and replace with one or two new issues, which I will do today.  (BTW I have not forgotten the `ApplicatorMixin` vs `Subschema` issue that I need to file).

Anyway, I will fix the changelog!